massa-hash/src/hash.rs

```diff
@@ -37,6 +37,22 @@ impl std::fmt::Debug for Hash {
     }
 }
 
+impl BitXor for Hash {
+    type Output = Self;
+
+    fn bitxor(self, other: Self) -> Self {
+        let xored_bytes: Vec<u8> = self
+            .to_bytes()
+            .iter()
+            .zip(other.to_bytes())
+            .map(|(x, y)| x ^ y)
```

Comment:
Since `u8` implements copy, `.map(|(x, y)| *x ^ *y)`, `.map(|(&x, &y)| x ^ y)` and `.map(|(x, y)| x ^ y)` all do the same thing. The question of the best practice might be worth asking but in terms of behaviour it's the same.

Reply:
I just found it more explicit and thus easier to read. As you prefer :)

Reply:
I personally prefer avoiding useless dereferencing but your point stands too